src/lib.rs

```diff
@@ -177,31 +175,37 @@ pub fn listen_fds() -> io::Result<u32> {
     let listen_pid = if let Ok(pid) = env::var("LISTEN_PID") {
         pid
     } else {
-        return Ok(0);
+        return Ok(0..0);
     }
     .parse::<u32>()
     .map_err(|_| io::Error::new(ErrorKind::InvalidInput, "invalid LISTEN_PID"))?;
 
     if listen_pid != process::id() {
-        return Ok(0);
+        return Ok(0..0);
     }
 
     let listen_fds = if let Ok(fds) = env::var("LISTEN_FDS") {
         fds
     } else {
-        return Ok(0);
+        return Ok(0..0);
     }
     .parse::<u32>()
     .map_err(|_| io::Error::new(ErrorKind::InvalidInput, "invalid LISTEN_FDS"))?;
 
+    let overflow = || io::Error::new(ErrorKind::InvalidInput, "fd count overflowed");
+
+    const SD_LISTEN_FDS_START: u32 = 3;
     let last = SD_LISTEN_FDS_START
         .checked_add(listen_fds)
-        .ok_or_else(|| io::Error::new(ErrorKind::InvalidInput, "fd count overflowed"))?;
+        .ok_or_else(overflow)?;
 
     for fd in SD_LISTEN_FDS_START..last {
         fd_cloexec(fd)?
     }
 
+    // let last = last.try_into::<i32>().map_err(overflow)?;
+    let last = i32::try_from(last).map_err(|_| overflow())?;
+    let listen_fds = SD_LISTEN_FDS_START as i32..last;
```

Comment:
Maybe it is safer to use `RawFd` directly here instead of `i32` (and same for `SD_LISTEN_FDS_START`)? That is just in case the `c_int` type is implemented as something other than `i32` on some architecture. Although that would probably not be a problem in most cases.

Reply:
I've replaced it with `RawFd`, though I doubt it matters too much. `SD_LISTEN_FDS_START` is still `u32` to keep it matched with the count. I suppose there's no need to go for `usize` yet.